### PR TITLE
DATA-1887 Convert Board to new proto response format

### PR DESCRIPTION
**Pull request by vijayvuyyuru**
This change is to change the board analog and GPIO pins to return single pins based on a pin name thats passed in via the config. I split this change out as the other changes were all identical and I didn't want this lost in there. The collector is per pin instead of returning all pins that are supplied.
**Testing**
Still working on testing, this will be tested manually and tests will be added as part of [DATA-2012](https://viam.atlassian.net/browse/DATA-2012)

[DATA-2012]: https://viam.atlassian.net/browse/DATA-2012?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/board/collector.go [hunk 1]
```diff
@@ -82,22 +60,21 @@ func newGPIOCollector(resource interface{}, params data.CollectorParams) (data.C
...
-				value, err := gpio.Get(ctx, data.FromDMExtraMap)
-				if err != nil {
-					// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
-					// is used in the datamanager to exclude readings from being captured and stored.
-					if errors.Is(err, data.ErrNoCaptureToStore) {
-						return nil, err
-					}
-					return nil, data.FailedToReadErr(params.ComponentName, gpios.String(), err)
+		var value bool
+		if gpio, err := board.GPIOPinByName(arg["reader_name"].String()); err == nil {
```
Comment (dmhilly) [inline components/board/collector.go:64]: Maybe pin_name or pin_number? Not actually sure what the input is here

**Review by dmhilly — COMMENTED** [review]
Just one comment - will re-review when you add those tests!

File: components/board/collector.go [hunk 2]
```diff
@@ -26,51 +27,28 @@ func (m method) String() string {
...
-				value, err := reader.Read(ctx, data.FromDMExtraMap)
-				if err != nil {
-					// A modular filter component can be created to filter the readings from a component. The error ErrNoCaptureToStore
-					// is used in the datamanager to exclude readings from being captured and stored.
-					if errors.Is(err, data.ErrNoCaptureToStore) {
-						return nil, err
-					}
-					return nil, data.FailedToReadErr(params.ComponentName, analogs.String(), err)
+		var value int
+		if reader, ok := board.AnalogReaderByName(arg["reader_name"].String()); ok {
```
Comment (AaronCasas) [inline components/board/collector.go:38]: Just checking - does that `.String()` call gracefully handle if there is no "reader_name" in the map?

Comment (vijayvuyyuru) [reply components/board/collector.go:38]: Will need to check this, I actually don't know.

File: components/board/collector.go (hunk 1 above)
Comment (vijayvuyyuru) [reply components/board/collector.go:64]: When I talked to Alexa, we settled on "reader_name": ____ but I also would be fine with pin_name.

**Comment by vijayvuyyuru** [comment]
Closing this pr to have it merge into the rest of collectors instead of main.

File: components/board/collector.go (hunk 1 above)
Comment (dmhilly) [reply components/board/collector.go:64]: SGTM, just wanted to make sure this decision was intentional. Thanks!

**Pull request closed without merging**